crates/ethrpc/src/block_stream/mod.rs

```diff
@@ -285,10 +287,20 @@ impl BlockRetrieving for Web3 {
                     serde_json::from_value::<web3::types::Block<H256>>(response.clone())
                         .with_context(|| format!("unexpected response format: {response:?}"))
                         .and_then(|response| {
-                            Ok((
-                                response.number.context("missing block number")?.as_u64(),
-                                response.hash.context("missing hash")?,
-                            ))
+                            let current_hash = response.hash.context("missing hash")?;
+                            let current_block = response.number.context("missing number")?.as_u64();
+                            if prev_hash.is_some_and(|prev| prev != response.parent_hash) {
+                                tracing::debug!(
+                                    ?range,
+                                    ?prev_hash,
+                                    parent_hash = ?response.parent_hash,
+                                    "detected reorg in block range"
```

Comment:
nanonit: technically it's not really that there is a reorg in the block range (we cannot tell that from looking at one sub-chain)

```suggestion
                                    "inconsistent parent in block range"
```